frontend/src/components/TimeRangeSelect.vue

```diff
@@ -1,58 +1,80 @@
 <template>
-    <v-select outlined
-              hide-details
-              dense
-              label="Timerange"
-              item-value="value"
-              item-text="label"
-              :items="options"
-              clearable
-              v-bind="$attrs"
-              v-on="$listeners"
+    <v-select
+      outlined
+      hide-details
+      dense
+      label="Timerange"
+      item-value="value"
+      item-text="label"
+      :items="options"
+      clearable
+      v-bind="$attrs"
+      v-on="$listeners"
     />
 </template>
 
 <script lang="ts">
 import Vue from 'vue'
 
+interface BuilderConfig {
+  seconds: number[];
+  minutes: number[];
+  hours: number[];
+  days: number[];
+}
+
+type BuilderConfigKeys = 'seconds' | 'minutes' | 'hours' | 'days'
+
+interface Option {
+  label: string;
+  value: number;
+}
+
 export default Vue.extend({
   inheritAttrs: false,
   data () {
     return {
-      options: [
-        {
-          label: 'last 15 minutes',
-          value: 15 * 60000
-        },
-        {
-          label: 'last 30 minutes',
-          value: 30 * 60000
-        },
-        {
-          label: 'last hour',
-          value: 60 * 60000
-        },
-        {
-          label: 'last 6 hours',
-          value: 6 * 60 * 60000
-        },
-        {
-          label: 'last 12 hours',
-          value: 12 * 60 * 60000
-        },
-        {
-          label: 'last 24 hours',
-          value: 24 * 60 * 60000
-        },
-        {
-          label: 'last 2 days',
-          value: 2 * 24 * 60 * 60000
+      builderConfig: {
+        minutes: [15, 30],
+        hours: [1, 6, 12, 24],
+        days: [2, 7]
+      },
+      options: [] as Option[]
+    }
+  },
+  created () {
+    this.options = this.buildTimeFilters()
+  },
+  methods: {
+    buildTimeFilters (): Option[] {
+      const additional = {
+        minutes: 1,
+        hours: 60,
+        days: 24 * 60
+      }
+
+      const config = this.builderConfig as BuilderConfig
```

Comment:
If you look into the other components I use the generic types for `Vue.extend` to define the interfaces. So you don't need the `as` in your actual code.

```ts
Vue.extend<Data, Methods, Computed, Props>(...)
```

Reply:
As suggested, I tried to implement the signature as shown below.

```
interface Data {
  builderConfig: BuilderConfig
  options: Option[]
  }

interface Methods {
  buildTimeFilters(): Option[]
}

interface Options {
  inheritAttrs: boolean;
}

export default Vue.extend<Data, Methods, null, Options>({...
```

However this generates three errors in the template section: 
```
 :items="options"
      v-bind="$attrs"
      v-on="$listeners"
```

for each of the former values I got the error: "Property 'options' does not exist on type 'never'"